src/main/java/com/dumbdogdiner/stickyapi/bukkit/item/PlayerHeadBuilder.java

```diff
@@ -0,0 +1,168 @@
+        playerId = player.getUniqueId();
+    }
+
+    /**
+     * @param player The Offline Player to use for head generation
+     */
+    public PlayerHeadBuilder(@NotNull OfflinePlayer player) {
+        this.meta.setOwningPlayer(player);
+        this.ownerProfile = Bukkit.createProfile(player.getUniqueId());
+        this.playerId = player.getUniqueId();
+    }
+
+    /**
+     * Creates a {@link PlayerHeadBuilder} from an existing head
+     *
+     * @param head an existing playerhead
+     * @throws IllegalArgumentException if the {@link ItemStack} is not a head, has bad metadata, or the profile attached is invalid
+     */
+    public PlayerHeadBuilder(@NotNull ItemStack head) {
+        Preconditions.checkArgument(head.getType() != Material.PLAYER_HEAD
+                        && head.getType() != Material.PLAYER_WALL_HEAD,
+                "Head must be a player head or player wall head");
+        meta = (SkullMeta) head.getItemMeta();
+        Preconditions.checkNotNull(meta, "Player head must have metadata attached");
+        // We check this almost immediately after, so it's OK if it's null temporarily
+        //noinspection ConstantConditions
+        ownerProfile = meta.getPlayerProfile();
+        name(meta.getDisplayName());
+        Preconditions.checkNotNull(ownerProfile, "The player head must have a PlayerProfile attached");
+        if (!Objects.requireNonNull(ownerProfile).hasTextures()) {
+            if (!ownerProfile.complete()) {
+                throw new IllegalArgumentException("Invalid player profile attached to the head, with no UUID or textures!");
+            }
+        }
+        this.playerId = null;
+    }
+
+    /**
+     * Statically sets the texture of the head so it will not update in the future
+     */
+    public @NotNull PlayerHeadBuilder freeze() {
+        if (playerId != null) {
+            String textureString = CachedMojangAPI.getTextureString(playerId);
+            assert textureString != null;
+            super.texture(textureString);
+            frozen = true;
+        }
+        return this;
+    }
+
+    /**
+     * Constructs a new {@link ItemStack} of type {@link Material#PLAYER_HEAD} based off the specified player
+     *
+     * @return A new player head {@link ItemStack}
+     */
+    @Override
+    public @NotNull ItemStack build() {
+        meta.setDisplayName(Objects.requireNonNullElseGet(name, () -> CachedMojangAPI.getUsername(playerId) + "'s Head"));
+
+        if (frozen) {
+            ownerProfile.setProperty(new ProfileProperty("texture", texture));
+        }
+        @NotNull ItemStack head = new ItemStack(Material.PLAYER_HEAD, quantity);
+        head.setItemMeta(meta);
+
+        return head;
+    }
+
+
+    // Disable methods of superclass:
+
+    /**
+     * This is unsupported.
+     *
+     * @throws UnsupportedOperationException if ran
+     */
+    @DoNotCall
+    @Override
+    @Deprecated
+    public @NotNull SkullBuilder texture(@NotNull String str) {
```

Comment:
See comment from before about hiding strictly internal methods - would be nice if we could rework this to not need that.

Reply:
Although, I do understand if that's too difficult.